### ISSUE TITLE
feature request - emit_soon() - alternate emit method for async

### ISSUE THREAD

**Issue by skewty** [riga/pymitter#10]
```python
class EventEmitter(object):
    def emit_soon(self, event: str, *args, **kwargs) -> None:
        """Alternate async handling. Similar to emit but awaitables are scheduled in current event loop but not run. They will be run the next time an await operation occurs.
        Use await asyncio.sleep(0) for them to run immediately or just use emit_async()."""
        awaitables = self._emit(event, *args, **kwargs)
        if awaitables:
            asyncio.ensure_future(asyncio.gather(*awaitables))

```

**Comment by riga** [comment]
Thanks for the feature request and the suggested implementation!

I called it `emit_future` to reflect that `asyncio.ensure_future` is used internally.

**Comment by riga** [comment]
Released [v0.4.2](https://pypi.org/project/pymitter/) with that change.

**Comment by skewty** [comment]
Perfect! Thank you very much for the super fast merge, version update and release.

I am already making use of it as:

```python
import asyncio
import warnings
import pymitter


class AsyncEventEmitter(pymitter.EventEmitter):
    """The default implementation of *emit()* will raise a *"RuntimeError: asyncio.run() cannot be called from a
    running event loop"* exception when called by a typical async application. This class variation has a safer
    implementation of the *emit()* method that should not raise those *RuntimeError* exceptions.
    """

    def emit(self, event: str, *args, **kwargs) -> None:
        """
        Emits an *event*. All functions of events that match *event* are invoked with *args* and
        *kwargs* in the exact order of their registration, if async functions are detected they
        are scheduled on the current running event loop (same as *emit_future* method) but with
        a *RuntimeWarning*.
        """

        if async_handlers := self._emit(event, *args, **kwargs):
            warnings.warn(
                "call to emit() with async handlers"
                "; operating as if emit_future() was used"
                "; use emit_async() or emit_future() directly to avoid this warning",
                category=RuntimeWarning,
                stacklevel=2,
            )
            asyncio.ensure_future(asyncio.gather(*async_handlers))
```

Including my usage here as I expect others to run into similar issues as I did.
Feel free to include this in the docs and in a future release for others if you feel it is of sufficient value.

Thanks again for this python module.  Love the wildcard support!
